XiEditor/XiCore.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+protocol XiCore {
+    func clientStarted(configDir: String?, clientExtrasDir: String?)
+    func setTheme(themeName: String)
+    func tracingConfig(enabled: Bool)
+}
+
+class CoreRPC: XiCore {
```

Comment:
will it be possible to subclass `CoreRPC`?

if not, you could annotate it with `final` and let the compiler does some optimizations.

what do you think?

Reply:
Thanks 👍